integration_tests/commands/http/check_type_test.go

```diff
@@ -0,0 +1,125 @@
+				{Command: "INCR", Body: map[string]interface{}{"key": "foo"}},
+			},
+			expected:   []interface{}{float64(1), setErrorMsg},
+			assertType: []string{"equal", "equal"},
+			delays:     []time.Duration{0, 0},
+		},
+		{
+			name: "DECR a key holding a set",
+			commands: []HTTPCommand{
+				{Command: "SADD", Body: map[string]interface{}{"key": "foo", "member": "bar"}},
+				{Command: "DECR", Body: map[string]interface{}{"key": "foo"}},
+			},
+			expected:   []interface{}{float64(1), setErrorMsg},
+			assertType: []string{"equal", "equal"},
+			delays:     []time.Duration{0, 0},
+		},
+		{
+			name: "BIT operations on a key holding a set",
+			commands: []HTTPCommand{
+				{Command: "SADD", Body: map[string]interface{}{"key": "foo", "member": "bar"}},
+				{Command: "GETBIT", Body: map[string]interface{}{"key": "foo", "offset": 1}},
+				{Command: "BITCOUNT", Body: map[string]interface{}{"key": "foo"}},
+			},
+			expected:   []interface{}{float64(1), setErrorMsg, setErrorMsg},
+			assertType: []string{"equal", "equal", "equal"},
+			delays:     []time.Duration{0, 0, 0},
+		},
+		{
+			name: "GETEX a key holding a set",
+			commands: []HTTPCommand{
+				{Command: "SADD", Body: map[string]interface{}{"key": "foo", "member": "bar"}},
+				{Command: "GETEX", Body: map[string]interface{}{"key": "foo"}},
+			},
+			expected:   []interface{}{float64(1), setErrorMsg},
+			assertType: []string{"equal", "equal"},
+			delays:     []time.Duration{0, 0},
+		},
+		{
+			name: "GETSET a key holding a set",
+			commands: []HTTPCommand{
+				{Command: "SADD", Body: map[string]interface{}{"key": "foo", "member": "bar"}},
+				{Command: "GETSET", Body: map[string]interface{}{"key": "foo", "value": "bar"}},
+			},
+			expected:   []interface{}{float64(1), setErrorMsg},
+			assertType: []string{"equal", "equal"},
+			delays:     []time.Duration{0, 0},
+		},
+		{
+			name: "LPUSH, LPOP, RPUSH, RPOP a key holding a set",
+			commands: []HTTPCommand{
+				{Command: "SADD", Body: map[string]interface{}{"key": "foo", "member": "bar"}},
+				{Command: "LPUSH", Body: map[string]interface{}{"key": "foo", "value": "bar"}},
+				{Command: "LPOP", Body: map[string]interface{}{"key": "foo"}},
+				{Command: "RPUSH", Body: map[string]interface{}{"key": "foo", "value": "bar"}},
+				{Command: "RPOP", Body: map[string]interface{}{"key": "foo"}},
+			},
+			expected:   []interface{}{float64(1), setErrorMsg, setErrorMsg, setErrorMsg, setErrorMsg},
+			assertType: []string{"equal", "equal", "equal", "equal", "equal"},
+			delays:     []time.Duration{0, 0, 0, 0, 0},
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			exec.FireCommand(HTTPCommand{Command: "DEL", Body: map[string]interface{}{"key": "foo"}})
+			for i, cmd := range tc.commands {
+				if tc.delays[i] > 0 {
+					time.Sleep(tc.delays[i])
+				}
+				result, _ := exec.FireCommand(cmd)
+				if len(tc.assertType) == 0 {
```

Comment:
Thanks for taking up my suggestion. Yes, the code format for the file was not as per the requirements, but there are lot of other PRs with the same change, I am sure one of those will have it fixed before this is merged. I have question with this` if ` condition, given that none of the test cases have an empty array.
Thanks

Reply:
Thanks for the clarification. Let me remove that if here also. Thanks for pointing out.

Reply:
I will still be keeping the switch case for the asserts for furture use cases.